les10 commit_HashMap

renvema opened the pull request:
Have a nice day)

Commit:
les10 commit

Sofasmile commented on src/task/tenth/MyHashMap.java:
```diff
@@ -0,0 +1,118 @@
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private Node<K, V>[] entries = new Node[DEFAULT_INITIAL_CAPACITY];
```
try to move initialization in the method put()

does we need @SuppressWarnings() here?

Sofasmile commented on src/task/tenth/MyHashMap.java:
```diff
@@ -0,0 +1,118 @@
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private void grow() {
+        Node<K, V>[] temp = (Node<K, V>[]) new Node[2 * entries.length];
```
does we need it? (Node<K, V>[])

Sofasmile commented on src/task/tenth/MyHashMap.java:
```diff
@@ -0,0 +1,118 @@
+                    temp[newIndex] = node;
+                    node = node.next;
+                } else {
+                    Node<K, V> nodeSecond = temp[newIndex];
```
better rename to currentNode

renvema replied:
done

Sofasmile commented on src/task/tenth/MyHashMap.java:
```diff
@@ -0,0 +1,118 @@
+                    Node<K, V> nodeSecond = temp[newIndex];
+                    while (nodeSecond != null) {
+                        if (nodeSecond.next == null) {
+                            nodeSecond.next = node;
```
we need add checking key.equals(entry.next.getKey()) here

renvema replied:
Sorry, I don't understand. It is metod grow, and it overwrite old values with new index.  It can't repeat keys, because I set new value in metod put, when keys are equals.  line24-26

Sofasmile commented on src/task/tenth/MyHashMap.java:
```diff
@@ -0,0 +1,118 @@
+            entries[index] = newNode;
+        } else {
+            Node<K, V> entry = entries[index];
+            if ((entry.getKey() == null && key == null) || entry.getKey().equals(key)) {
```
better divide this if statement

Sofasmile commented on src/task/tenth/MyHashMap.java:
```diff
@@ -0,0 +1,118 @@
+            }
+            entry = entry.next;
+            while ((entry) != null) {
+                if (key.equals(entry.next.getKey())) {
```
```suggestion
                if (key.equals(entry.getKey())) {
```

Sofasmile replied:
we can remove 30 line to make this method work correctly

Sofasmile commented on src/task/tenth/MyHashMap.java:
```diff
@@ -0,0 +1,118 @@
+            entry = entry.next;
+            while ((entry) != null) {
+                if (key.equals(entry.next.getKey())) {
+                    entry.next.setValue(value);
```
```suggestion
                    entry.setValue(value);
```

Commit:
les10 commit2
les10 commit2
- move initialization in the method put();
- remove  @SuppressWarnings() (I used it. because line colored yellow and German used it on lessons)
- remove (Node<K, V>[]);
-rename to currentNode;

-don't add checking key.equals(entry.next.getKey()). I don't understand this comment. It is method grow, and it overwrite old values with new index. It can't repeat keys, because I set new value in method put, when keys are equals. line24-26
